lib/ui/lerp.dart

```diff
@@ -6,15 +6,23 @@
 
 part of dart.ui;
 
-/// Linearly interpolate between two numbers.
-// TODO(cbracken): Consider making a and b non-nullable.
-//                 https://github.com/flutter/flutter/issues/64617
+/// Linearly interpolate between two numbers, `a` and `b`, by an extrapolation
+/// factor `t`.
+///
+/// When both `a` and `b` are null, infinite, or NaN, `a` is returned.
+/// Otherwise, if `a`, `b`, and `t` are required to be finite or null, and the
+/// result of `a + (b - a) * t` is returned, where nulls are defaulted to 0.0.
 double? lerpDouble(num? a, num? b, double t) {
-  if (a == null && b == null)
-    return null;
-  a ??= 0.0;
-  b ??= 0.0;
-  return a + (b - a) * t as double;
+  if (a == b)
+    return a as double?;
+  final double from = a as double? ?? 0.0;
+  final double to = b as double? ?? 0.0;
+  if (from.isNaN && to.isNaN)
```

Comment:
i think this would be clearer if it was above the from/to declarations (and used a/b)

Reply:
Did that initially, but it gets a bit uglier since we'd need `as double?` casts on them to get `isNaN`. Can switch it back if you think that's better.

Reply:
Ignore that, I'd confused myself apparently. Done.